shared/reducers/fs.js

```diff
@@ -172,13 +150,15 @@ export default function(state: Types.State = initialState, action: FsGen.Actions
       return state.mergeIn(['flags'], {kbfsInstalling: true})
     case FsGen.localHTTPServerInfo:
       return state.set('localHTTPServerInfo', Constants.makeLocalHTTPServer(action.payload))
-    case FsGen.favoriteIgnore:
-      return state.mergeIn(['pathItems', Types.pathToString(action.payload.path), 'tlfMeta'], {
-        isIgnored: true,
-      })
+    case FsGen.favoriteIgnore: // fallthrough
     case FsGen.favoriteIgnoreError:
-      return state.mergeIn(['pathItems', Types.pathToString(action.payload.path), 'tlfMeta'], {
-        isIgnored: false,
+      const elems = Types.getPathElements(action.payload.path)
+      const visibility = Types.getVisibilityFromElems(elems)
+      if (!visibility) {
+        return state
+      }
+      return state.mergeIn(['tlfs', visibility, elems[2]], {
+        isIgnored: action.type === FsGen.favoriteIgnore,
```

Comment:
I don't think this is right--it could be that one `favoriteIgnore` succeeded and then another one failed, in which case the failing one would reset the value of `isIgnored`.

Reply:
Hmm I think this is equivalent to the version before. I don't think it really matters much though. `isIgnore` is just for update the UI state right? On the next loadFavorites call, this won't be part of it if ignore succeeded.

Reply:
Yeah true.